### ISSUE TITLE
Arduino Code support.

### ISSUE THREAD

**Issue by ouTPhaze** [fritzing/fritzing-app#4038]
## Problem
Simulating any Arduino Application is currently impossible.

## Proposed Solution
Allow the capability to simulate Arduino Code within the program, this could do wonders when prototyping.

For an example of Arduino Code simulation in action, check out: https://tinkercad.com

Thank you for your time.

**Comment by KjellMorgenstern** [comment]
If you want this request to be taken serious, a way more detailed ticket is needed (my guess chances to simulate Arduino in near future are dim, but without a detailed look, we won't know)

While this is a very common request, does it really solve a problem? "Sketching in hardware" is a real thing. Do you really learn electronics when pressing a "Play" button?

How many people would want to use that simulator?

By our current calculation, this would be quite expensive to develop. How many free Arduinos could we give out to everyone who wanted a simulator? 

How accurate does the simulator need to be?

Do you just want to see some movement and blinking on the screen? Or do you want the exact same problems, that a real circuit could show?

If you use the simulator, do you want to run diagnostics about your Arduino code running? Or your circuit?
Or have you already checked these separately, and now want to verify the integration of both?

How much effort would that be, what is your estimate? (Tip: We won't know until we have better requirements)

Are there existing techs we can use, and how?

Only Arduino UNO ? Or also the new R4? 

Please add more questions like that, just getting started here ;-) Answers not yet needed.

**Comment by SasmeetNadkarni** [comment]
An Arduino  simulator would be wonderful.  Was thinking about teaching kids  about electronics and this is the perfect software.
It is  offline and the new simulator is also good.

**Comment by KjellMorgenstern** [comment]
This looks like one of the most requested features, but we need numbers, as it would be quite an investment. Please use the "Thumbs up" feature of github :+1:

**Comment by RAPTOR7762** [comment]
Yes, it's like building another fritzing. There are also many other types of arduinos around...
